draft-ietf-ppm-dap.md

```diff
@@ -1214,64 +1269,174 @@ as follows:
+1. An output share, in which case the Helper stores the output share for future
+   collection as described in {{collect-flow}} and replies to the Leader with a
+   `PrepareStep` in the `finished` state. The Helper expects no further requests
+   from the Leader for a report in the `finished` state.
+1. An updated VDAF state and preparation message, denoted
+   `(prep_state, prep_msg)`, in which case the Helper replies to the leader with
+   `PrepareStep` in the `continued` state containing `prep_msg`.
 
-The message is a sequence of PrepareStep values, the order of which matches that
-of the ReportShare values in `AggregateInitializeReq.report_shares`. Each report
-that was marked as invalid is assigned the PrepareStepResult `failed`.
-Otherwise, the PrepareStep is either marked as continued with the output
-`prep_msg`, or is marked as finished if the VDAF preparation process is finished
-for the report share.
+After computing each report share's preparation state and prepare message, the
+Helper SHOULD store them, tagged with the round they were computed in, so that
+the Leader can recover from losing the Helper's response.
 
-The helper's response to the leader is an HTTP status code 200 OK whose body is
-the AggregateInitializeResp and media type is
-"application/dap-aggregate-initialize-resp".
+If the `round` in the Leader's request is equal to the Helper's current round
+(i.e., this is not the first time the Leader has sent this request), then the
+Helper can either re-compute the current round prepare message or re-transmit a
+previously computed message. The Helper MUST verify that the contents of the
+`AggregationJob` are identical to the previous message (i.e., containing the
+same set of reports).
 
-Upon receipt of a helper's AggregateInitializeResp message, the leader checks
-that the sequence of PrepareStep messages corresponds to the ReportShare
-sequence of the AggregateInitializeReq. If any message appears out of order, is
-missing, has an unrecognized report ID, or if two messages have the same report
-ID, then the leader MUST abort with error "unrecognizedMessage".
+If the Leader's `round` is behind or more than one round ahead of the Helper's
+current round, then the Helper MUST abort with an error of type `roundMismatch`.
 
-[[OPEN ISSUE: the leader behavior here is sort of bizarre -- to whom does it
-abort?]]
+The Helper's response to the Leader is an `AggregationJob` (see
+{{aggregation-job-representation}}) compiled from the `PrepareStep`s. The order
+of the Helper's `prepare_steps` MUST match that of the Leader's.
+
+[[OPEN ISSUE: consider relaxing this ordering constraint. See issue#217.]]
+
+##### DELETE `/tasks/{task-id}/aggregation_jobs/{aggregation-job-id}` {#aggregation-job-delete}
+
+Indicates to the Helper that it MAY abandon the aggregation job and discard all
+state related to it. Whether or not it deletes any data, the Helper MUST respond
```

Comment:
I think this text should allow returning e.g. `500` on error -- as written, the Helper must unconditionally respond with a `204`.

Reply:
- [ ] clean up language about HTTP status code in {{aggregation-job-delete}}